core/src/test/java/test/failedreporter/issue2517/FailedDataProviderWithFactoryReporterTest.java

```diff
@@ -0,0 +1,75 @@
+package test.failedreporter.issue2517;
+
+import org.testng.Assert;
+import org.testng.TestNG;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.collections.Lists;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+import test.SimpleBaseTest;
+
+import java.io.File;
+import java.util.List;
+
+public class FailedDataProviderWithFactoryReporterTest extends SimpleBaseTest {
+  private File mTempDirectory;
+
+  @BeforeMethod
+  public void setUp() {
+    mTempDirectory = createDirInTempDir("testng-tmp-" + System.currentTimeMillis() % 1000);
+  }
+
+  @AfterMethod
+  public void tearDown() {
+    deleteDir(mTempDirectory);
+  }
+
+  @Test
+  public void failedMethodWithDataProviderShouldHaveInvocationNumbers() {
+    testFailedReporter(
+        new String[] {"f1"},
+        "<include name=\"%s\" invocation-numbers=\"1\"/>",
+        FailedDataProviderWithFactoryReporterSampleTest.class);
+  }
+
+  private void triggerTest(String packageName, boolean applyGroupSelectionAtTest) {
```

Comment:
I think the method is not used. Could you remove it?

Reply:
done